Networking/Networking/Network/WordPressOrgNetwork.swift

```diff
@@ -0,0 +1,142 @@
+import Alamofire
+import Combine
+import Foundation
+import WordPressKit
+
+/// Class to handle WP.org REST API requests.
+///
+public final class WordPressOrgNetwork: Network {
+
+    private let authenticator: Authenticator?
+    private let userAgent: String?
+
+    private lazy var sessionManager: Alamofire.SessionManager = {
+        let sessionConfiguration = URLSessionConfiguration.default
+        let sessionManager = makeSessionManager(configuration: sessionConfiguration)
+        return sessionManager
+    }()
+
+    private lazy var backgroundSessionManager: Alamofire.SessionManager = {
+        // A unique ID is included in the background session identifier so that the session does not get invalidated when the initializer is called multiple
+        // times (e.g. when logging in).
+        let uniqueID = UUID().uuidString
+        let sessionConfiguration = URLSessionConfiguration.background(withIdentifier: "com.automattic.woocommerce.backgroundsession.\(uniqueID)")
+        let sessionManager = makeSessionManager(configuration: sessionConfiguration)
+        return sessionManager
+    }()
+
+    public init(authenticator: Authenticator? = nil, userAgent: String? = nil) {
+        self.authenticator = authenticator
+        self.userAgent = userAgent
+    }
+
+    public func responseData(for request: URLRequestConvertible) async throws -> Data? {
+        return try await withCheckedThrowingContinuation { [weak self] continuation in
+            guard let self = self else { return }
+
+            self.sessionManager.request(request)
+                .validate()
+                .responseData(completionHandler: { (response) in
+                switch response.result {
+                case .success(let responseObject):
+                    continuation.resume(returning: responseObject)
+                case .failure(let error):
+                    DDLogWarn("⚠️ Error requesting \(request.urlRequest?.url?.absoluteString ?? ""): \(error.localizedDescription)")
+                    continuation.resume(throwing: error)
+                }
+
+            })
+        }
+    }
+
+    /// Executes the specified Network Request. Upon completion, the payload will be sent back to the caller as a Data instance.
+    ///
+    /// - Important:
+    ///     - User agent and authenticator from the initializer will be injected.
+    ///
+    /// - Parameters:
+    ///     - request: Request that should be performed.
+    ///     - completion: Closure to be executed upon completion.
+    ///
+    /// - Note:
+    ///     - The response body will always be returned (when possible), even when there's a networking error.
+    ///       This differs slightly from the standard Alamofire `.validate()` behavior, and it's required so that
+    ///       the upper layers can properly detect "Jetpack Tunnel" Errors.
+    ///     - Yes. We do the above because the Jetpack Tunnel endpoint doesn't properly relay the correct statusCode.
+    ///
```

Comment:
> Yes. We do the above because the Jetpack Tunnel endpoint doesn't properly relay the correct statusCode.

❓ Is this comment valid here?

Reply:
Updated comments for `WordPressOrgNetwork` in 0c5b2374329363b44cb0b476b43e9d76f1652759.